pkgs/by-name/or/orbstack/package.nix

```
@@ -0,0 +1,83 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  _7zz,
+}:
+let
+  inherit (stdenvNoCC.hostPlatform) system;
+  version = "2.0.3-19876";
+  sourceData = {
+    aarch64-darwin = {
+      arch = "arm64";
+      hash = "sha256-3Ppc0zWEgR/nTS7R9uAkUYYgYu5q2TWmfd3evT+Z8g4=";
+    };
+    x86_64-darwin = {
+      arch = "amd64";
+      hash = "sha256-+JpyynFKmDjHLetvvEpQ0qw4crAVmx0ucWm+bvtZ2Fg=";
+    };
+  };
+  sources = lib.mapAttrs (
+    system:
+    { arch, hash }:
+    fetchurl {
+      url = "https://cdn-updates.orbstack.dev/${arch}/OrbStack_v${
+        lib.replaceString "-" "_" version
+      }_${arch}.dmg";
+      inherit hash;
+    }
+  ) sourceData;
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "orbstack";
+  inherit version;
+
+  src = finalAttrs.passthru.sources.${system} or (throw "unsupported system ${system}");
+
+  # -snld prevents "ERROR: Dangerous symbolic link path was ignored"
+  # -xr'!*:com.apple.*' prevents macOS extended attributes (e.g. macl or
+  # quarantine) being turned into real files when extracting an APFS .dmg
+  # (e.g. Info.plist:com.apple.macl or Info.plist:com.apple.quarantine).
+  # These bogus files corrupt the .app bundle and prevent it from launching.
+  unpackCmd = "7zz x -snld -xr'!*:com.apple.*' $curSrc";
+
+  nativeBuildInputs = [ _7zz ];
+
+  sourceRoot = ".";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out/Applications"
+    cp -R OrbStack.app "$out/Applications"
+
+    mkdir -p "$out/bin"
+    for bindir in bin xbin; do
+      for binary in "$out/Applications/OrbStack.app/Contents/MacOS/$bindir"/*; do
+        if [[ -f "$binary" ]]; then
+          ln -s "$binary" "$out/bin/$(basename "$binary")"
+        fi
+      done
+    done
```

```suggestion
    for binary in "$out"/Applications/OrbStack.app/Contents/MacOS/{bin,xbin}/*; do
      ln -s "$binary" "$out/bin/$(basename "$binary")"
    done
```
How about this simpler variant?

---

That's much nicer, thank you.